Web/tues_pairs/src/components/Authentication/withAuthentication.jsx

```diff
@@ -25,7 +25,33 @@ const withAuthentication = Component => {
                             log.info("Successfully garnered current user information!");
 
                             this.setState({ authUser });
-                        });
+                        }).then(() => {
+                            this.props.firebase.messaging.requestPermission().then(() => {
+                                this.props.firebase.messaging.getToken().then(token => {
+                                    let deviceTokens;
+                                    if(authUser.deviceTokens) {
+                                        deviceTokens = authUser.deviceTokens;
+                                    } else {
+                                        deviceTokens = [];
+                                    }
+                                    if(!deviceTokens.includes(token)) {
+                                        deviceTokens.push(token);
+                                    }
+
+                                    this.props.firebase.db.collection("users").doc(authUser.uid).set({
+                                        deviceTokens: deviceTokens
+                                    }, { merge: true });
+
+                                    authUser.deviceTokens = deviceTokens;
+
+                                    this.setState({ authUser });
```

Comment:
Why not just use authUser.deviceTokens, or at least put it in a variable?

Reply:
Yes like that I will skip the use of the unneeded variable deviceTokens. I will simply use the authUser.deviceTokens to manipulate the array.